crates/edr_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -339,6 +397,16 @@ impl WriteCacheKey {
         Self::Resolved(hasher.finalize())
     }
 
+    fn needs_range_check(hasher: Hasher, range: CacheableLogFilterRange<'_>) -> Option<Self> {
+        match range {
+            CacheableLogFilterRange::Hash(_) => Some(Self::finalize(hasher)),
+            CacheableLogFilterRange::Range { to_block, .. } => {
+                // TODO should we check that to < from?
+                Self::needs_safety_check(hasher, to_block)
```

Comment:
I'm not sure what's the expected behavior if `to < from`, but to be safe, we could potentially check the safety of the max of `to` and `from`. Wdyt?

Reply:
For Alchemy, if `to < from`, you get this error:

```rust
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"One of the blocks specified in filter (fromBlock, toBlock or blockHash) cannot be found."}}
```

so using the maximum of those two wouldn't resolve the issue, I think. We might just have to prohibit caching if this occurs?